### PR TITLE
[05] Add plotting research funding vs private investment

### LINKED ISSUE #5: Add plotting for research funding vs investments

### PR DESCRIPTION
**Pull request by Jack-0-0**
Closes #5.

This PR adds:
* plotting functions into `utils/plotting.py` for plotting charts with two y axis and for creating lag plots
* using the above functions for analysis in `analysis/tech_category_level/research_funding_and_investment.py` notebook 
* a note into the readme on how to download inputs using `make inputs-pull`

### CODE REVIEW

File: iss_forecasting/analysis/utils/plotting.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+def plot_two_y_one_x(
+    data_source,
+    x,
+    y1,
+    y2,
+    chart_title,
+    line_1_colour="#5276A7",
+    line_2_colour="#57A44C",
+):
```
Comment (beingkk) [inline iss_forecasting/analysis/utils/plotting.py]: Hey, would it be OK to use type hints in the definitions of function? (as explained [here](https://docs.python.org/3/library/typing.html)). Do let me know if you have objection to this style.

For example:

```suggestion
def plot_two_y_one_x(
    data_source: pd.DataFrame,
    x: str,
    y1: str,
    y2: str,
    chart_title: str,
    line_1_colour: str="#5276A7",
    line_2_colour: str="#57A44C",
) -> altair.vegalite.v4.api.LayerChart:
```

Comment (beingkk) [reply iss_forecasting/analysis/utils/plotting.py:36]: For the `LayerChart` perhaps even makes sense to use a type alias (eg, if this is going to be the output type for multiple functions)

Comment (beingkk) [reply iss_forecasting/analysis/utils/plotting.py]: For the LayerChart perhaps even makes sense to use a type alias (eg, if this is going to be the output type for multiple functions)

File: iss_forecasting/analysis/utils/plotting.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    line_1 = base.mark_line(color=line_1_colour).encode(
+        alt.Y(y1, axis=alt.Axis(titleColor=line_1_colour))
+    )
+    line_2 = base.mark_line(color=line_2_colour).encode(
+        alt.Y(y2, axis=alt.Axis(titleColor=line_2_colour))
+    )
+    return alt.layer(line_1, line_2).resolve_scale(y="independent")
+
+
+def lagplot(x, y=None, lag=1, ax=None, **kwargs):
```
Comment (beingkk) [inline iss_forecasting/analysis/utils/plotting.py]: Same point about type hints

**Review by beingkk — APPROVED** [review]
Hey, thanks for this - looks good, just one suggestion to use [type hints](https://docs.python.org/3/library/typing.html) when defining functions. Do let me know if you have objections to that style though.

File: iss_forecasting/analysis/utils/plotting.py (hunk 1 above)
Comment (Jack-0-0) [reply iss_forecasting/analysis/utils/plotting.py]: Is this so it is more readable? If so, I have added `from altair.vegalite.v4.api import LayerChart` at the top of the file so just `LayerChart` can be used here.

**Comment by Jack-0-0** [comment]
> Hey, thanks for this - looks good, just one suggestion to use [type hints](https://docs.python.org/3/library/typing.html) when defining functions. Do let me know if you have objections to that style though.

Thank you for the review @beingkk and thanks for the suggestion of using type hints. I've updated all the functions to use type hints.

**Comment by beingkk** [comment]
Nice, well done! I didn't know about the `Optional` type, will try to use it as well.

**Pull request merged**